src/storage/src/storage_state.rs

```diff
@@ -220,6 +222,12 @@ impl<'w, A: Allocate> Worker<'w, A> {
                 self.timely_worker.step();
             }
 
+            // Rerport any dropped ids
+            if !self.storage_state.dropped_ids.is_empty() {
+                let ids = self.storage_state.dropped_ids.split_off(0);
```

Comment:
`split_off(0)` is a bit weird here, honestly `drain(..).collect()` or `mem::swap` make more sense

Reply:
I was trying to be smart and reuse the allocation. I'll use `std::mem::take`

Reply:
nice